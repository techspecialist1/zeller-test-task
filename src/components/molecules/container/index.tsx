import React, {FC} from 'react';
import {SafeAreaView, View} from 'react-native';

import {styles} from './styles';

type ContainerProps = {
  children: React.ReactNode;
};

const Container: FC<ContainerProps> = ({children}) => {
  return (
    <SafeAreaView style={styles.safeArea}>
      <View style={styles.container}>{children}</View>
    </SafeAreaView>
  );
};

export default Container;
