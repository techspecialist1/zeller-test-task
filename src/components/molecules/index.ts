import ListView from './listView';
import Container from './container';
import RadioButton from '../atoms/radioButton';
import SearchInput from '../atoms/searchInput';
import ActivityIndicator from '../atoms/activityIndicator';
import ErrorBox from '../atoms/errorBox';

export {
  ListView,
  ActivityIndicator,
  RadioButton,
  SearchInput,
  Container,
  ErrorBox,
};
