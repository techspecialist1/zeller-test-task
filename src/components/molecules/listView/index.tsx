import React from 'react';
import {View, Text, TouchableOpacity} from 'react-native';

import {styles} from './style';

type ListViewProps = {
  item: {
    name: string;
    role: string;
  };
  onPress: () => void;
};

const ListView: React.FC<ListViewProps> = props => {
  const {item, onPress} = props;

  return (
    <TouchableOpacity onPress={onPress}>
      <View style={styles.userList}>
        <View style={styles.profile}>
          <Text style={styles.profileText}>{item?.name?.slice(0, 1)}</Text>
        </View>
        <View style={styles.listName}>
          <Text style={styles.name}>{item?.name}</Text>
          <Text style={styles.role}>{item?.role}</Text>
        </View>
      </View>
    </TouchableOpacity>
  );
};

export default ListView;
