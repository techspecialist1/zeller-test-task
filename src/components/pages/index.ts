import HomePage from './homePage';
import DetailPage from './detailPage';

export {HomePage, DetailPage};
