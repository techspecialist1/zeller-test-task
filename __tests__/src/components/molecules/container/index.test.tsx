import React from 'react';
import renderer from 'react-test-renderer';
import {View} from 'react-native';
import {Container} from '../../../../../src/components/molecules';

describe('Container', () => {
  test('renders correctly with children', () => {
    const component = renderer.create(
      <Container>
        <View testID="child-view" />
      </Container>,
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('renders children correctly', () => {
    const component = renderer.create(
      <Container>
        <View testID="child-view" />
      </Container>,
    );
    const instance = component.root;
    const childView = instance.findByProps({testID: 'child-view'});

    expect(childView).toBeTruthy();
  });
});
