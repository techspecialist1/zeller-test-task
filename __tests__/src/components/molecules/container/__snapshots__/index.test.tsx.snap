// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container renders correctly with children 1`] = `
<RCTSafeAreaView
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    style={
      {
        "backgroundColor": "FFFFFF",
        "flex": 0.7,
        "padding": 19.692307692307693,
      }
    }
  >
    <View
      testID="child-view"
    />
  </View>
</RCTSafeAreaView>
`;
